thefuck/rules/apt_unable_to_locate.py

```diff
@@ -0,0 +1,55 @@
+import subprocess
+
+from thefuck.specific.sudo import sudo_support
+from thefuck.utils import for_app, replace_command
+
+
+@sudo_support
+@for_app("apt", "apt-get")
+def match(command):
+    return "E: Unable to locate package" in command.output
+
+
+def _parse_apt_search_results(search_text_lines):
+    search_results = []
+
+    for line in search_text_lines:
+        line = line.decode().strip()
+        if line.find("/") > 0:
+            search_results.append(line.split("/")[0])
+
+    return search_results
+
+
+def _parse_apt_cache_search_results(search_text_lines):
+    search_results = []
+
+    for line in search_text_lines:
+        line = line.decode().strip()
+        if line.find(" - ") > 0:  # spaces are important
+            search_results.append(line.split(" - ")[0])
+
+    return search_results
+
+
+def _get_search_results(app, command):
```

Comment:
apt_get.py uses CommandNotFound to achieve something similar to this (though I've not looked to see if it is identical); would it be prudent for this to do it the same way?  Or maybe have an apt_support wrapper like we do with git?

Reply:
Hey David, you are right! I have been feeling a little burned out right now, I will try to work on this Friday!

Reply:
I am looking into https://github.com/nvbn/thefuck/blob/master/thefuck/rules/apt_get.py now I will get back to you in a few days

Reply:
Before I forget, there's no memoizing here either.